### PR TITLE
fix: bg substraction for float images with OpenCL

### PR DESCRIPTION
**Pull request by matthieu-ft**
this is my dirty fix for issue #6044, I am open to suggestions about how to make it look nicer

### CODE REVIEW

**Comment by alalek** [comment]
Test with float mode is required to prove that patch works.

File: modules/video/src/bgfg_gaussmix2.cpp [hunk 1]
```diff
@@ -905,24 +898,47 @@ void BackgroundSubtractorMOG2Impl::getBackgroundImage(OutputArray backgroundImag
...
+{
+    CV_Assert(frameType == CV_8UC1 || frameType == CV_8UC3 || frameType == CV_32FC1 || frameType == CV_32FC3);
+
+#ifdef HAVE_OPENCL
+    if (opencl_ON)
+    {
+        CV_OCL_RUN(opencl_ON, ocl_getBackgroundImage(backgroundImage))
+
+        opencl_ON = false;
+        return;
```
Comment (alalek) [inline modules/video/src/bgfg_gaussmix2.cpp]: Why is "return" statement here? In case of failure we should call fallback CPU processing

**Comment by matthieu-ft** [comment]
Hi,
I added the required tests, and the build passed. Looking at the details, it seems that the build hasn't been tested on OpenCL Intel, Win32... is it normal behavior?

File: modules/video/src/bgfg_gaussmix2.cpp (hunk 1 above)
Comment (matthieu-ft) [inline modules/video/src/bgfg_gaussmix2.cpp]: Actually I didn't add this code, getBackgroundImage and the return where already there. I just added getBackgroundImage_intern before the method, and moved some of the content from getBackgroundImage to getBackgroundImage_intern, therefore it looks like getBackgroundImage is new.
I just looked at the definition of CV_OCL_RUN I saw that it does make a return when required, so indeed we don't need the return here. I remove it.

**Comment by matthieu-ft** [comment]
What should be the next step now? Is there anything else to be done before it gets merged?

**Comment by alalek** [comment]
Thanks!
:+1:

**Pull request merged**